### DISCUSSION TITLE
NodeToPython v3.0.0

### DISCUSSION THREAD

**Discussion by BrendanParmer** [BrendanParmer/NodeToPython#86] [Announcements]
NodeToPython v3.0 is finally here!

**Blender 4.0 is now officially supported.** NodeToPython has been upgraded to handle **Node Panels** and the new NodeTree interface, and it now supports **Repeat Zones**, **Geometry Node Tools**, the new **Rotation socket**, and a whole host of new nodes and settings

Joining its Geometry and Shader counterparts, NodeToPython can now generate **Compositor Nodes** for a scene. 

This update includes a bunch of other improvements, including
* More robust variable name creation
* Setting a node's `name` attribute
* Fixing issues with how subgroups were handled
* A major refactor, which cleaned up the code and should speed up future development

Check out the full list of changes [here](https://github.com/BrendanParmer/NodeToPython/pull/85#issue-2103959463)

Find a problem or have a suggestion? Please [submit an issue!](https://github.com/BrendanParmer/NodeToPython/issues/new/choose)

<hr /><em>This discussion was created from the release <a href='https://github.com/BrendanParmer/NodeToPython/releases/tag/v3.0.0'>NodeToPython v3.0.0</a>.</em>

**Comment by Dangry98** [comment]
Thanks a lot for the update! ❤️ 
Super usful addon! 🔥
